File: js/axios/api-keys/authenticateKey.js
import axios from "axios";
import "dotenv/config";

const accessKey = process.env.ACCESS_TOKEN;
const apiUrl = process.env.production
    ? "https://api.theauthapi.com"
    : process.env.TESTING_URL;

async function authenticateKey(apiKey) {
  try {
    const {data: authenticatedKey} = await axios
        .post(`${apiUrl}/api-keys/auth/${apiKey}`, {}, {
          headers: {
            ContentType: "application/json",
            access_key: accessKey,
          },
        })
    return authenticatedKey;
  } catch (error) {
    console.log(error);
    if (error.response) {
      if (error.response.status === 404) {
        // handle invalid key
      } else if (error.response.status === 429) {
        // handle api key rate limit error
      } else {
        // handle 401, 403 errors (these should not occur if you did setup your configs correctly
      }
    } else if (error.request) {
      // request was never sent to theauthapi, handle request error here
    } else {
      // handle other errors
    }
  }

}

(async () => {
  const authenticatedKey = await authenticateKey(process.env.TEST_KEY);
  console.log(authenticatedKey)
})();

File: python/api-keys/authenticate_key.py
import requests
from requests.exceptions import HTTPError

api_url = "https://api.theauthapi.com/api-keys"
access_key = "REPLACE_WITH_YOUR_ACCESS_KEY"

headers = {
    "x-api-key": access_key,
}

def authenticate_api_key(api_key):
    try:
        response = requests.post(f'{api_url}/auth/{api_key}', headers=headers)
        response.raise_for_status()
        return response.json()
    except HTTPError as error:
        if error.response.status_code == 404:
            # handle invalid key
            print("Invalid API key")
        else:
            # handle other error
            print(error)

key_to_authenticate = 'REPLACE_WITH_YOUR_API_KEY'
authenticated_key = authenticate_api_key(key_to_authenticate)
print(authenticated_key)
